### DISCUSSION TITLE
Why is FirebaseRecaptcha getting deprecated? What about web users?

### DISCUSSION THREAD

**Discussion by nubpro** [expo/expo#20149] [Archived: Questions]
I understand that [FirebaseRecaptcha](https://docs.expo.dev/versions/latest/sdk/firebase-recaptcha/#with-react-native-firebase) is getting deprecated, and moving forward, the recommendation is to install `React Native Firebase`. However, the suggested package does not support web OOTB. And the app we are building needs to work on the web and mobile platforms, hence what are the recourse for us?
Can we still stick to the the soon to be deprecated package?

**Comment by keith-kurak** [comment]
If Recaptcha is the only Firebase feature using, it's technically possible you'd be able to continue using it for sometime past SDK 48, for as long as there are no native compatibility issues between the SDK and the underlying native Firebase code. It's hard to predict if and/ or when such incompatibilities could arise, so we can't support this scenario.

For Recaptcha on web, [you'll want to switch to using Firebase JS SDK's implementation directly](https://firebase.google.com/docs/app-check/web/recaptcha-provider). If you use both React Native Firebase for native and Firebase JS for web, you can switch between the two by wrapping your code in platfrom-specific extensions, e.g., **recaptcha.web.js** and **recaptcha.native.js**. This was more-or-less the only function of the `expo-firebase-*` packages under EAS Build- to switch between native and web functions. By moving over to using these implementations directly, it will ensure developers are not blocked from using features supported by these SDK's going forward.

**Reply by nubpro** [reply]
Thanks for the explanation, I will continue using the package for now (I need to get things going)
But the problem I face now is that the [sample snack](https://docs.expo.dev/versions/latest/sdk/firebase-recaptcha/#basic-usage) provided by Expo is not working on the web (ps: you need to pull the project down locally)
We are hitting this error on the web `index.esm2017.js:230 Uncaught FirebaseError: Firebase: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp() (app-compat/no-app).` even though Firebase is initialized.

I'm looking into the source currently, but I'm not able to understand what is it lacking of. Appreciate it if you could give me some rope

**Reply by keith-kurak** [reply]
If you already have Recaptcha working, I would leave it as-is until you're ready to upgrade. If you're just starting to implement it, I would strongly recommend not using `expo-firebase-recaptcha`. Even if it doesn't break right away, at some point, it is likely to stop working, so you might as well spend your development effort using the Firebase libraries directly.

**Reply by nubpro** [reply]
> If you already have Recaptcha working, I would leave it as-is until you're ready to upgrade. If you're just starting to implement it, I would strongly recommend not using `expo-firebase-recaptcha`. Even if it doesn't break right away, at some point, it is likely to stop working, so you might as well spend your development effort using the Firebase libraries directly.

Yes I understand where you're coming from, but unfortunately having too much moving pieces at once is going to take a toll on our time and development effort. It's simply more straightforward to get the package working, with that said I've identified the problem and made a PR #20181
